File: enum-jpa-querydsl-app/src/main/java/com/example/enumapp/web/ApiErrorBody.java
package com.example.enumapp.web;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.util.Map;

/**
 * 오류 시 {@link ApiResponse#getResult()} 에 담기는 본문.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ApiErrorBody {

    private String message;
    private String messageCode;
    private Map<String, String> errors;

    public static ApiErrorBody of(String messageCode, String message, Map<String, String> errors) {
        ApiErrorBody body = new ApiErrorBody();
        body.messageCode = messageCode;
        body.message = message;
        body.errors = errors;
        return body;
    }

    public String getMessage() {
        return message;
    }

    public String getMessageCode() {
        return messageCode;
    }

    public Map<String, String> getErrors() {
        return errors;
    }
}

File: enum-jpa-querydsl-kotlin-app/src/main/kotlin/com/example/enumapp/web/ApiErrorBody.kt
package com.example.enumapp.web

import com.fasterxml.jackson.annotation.JsonInclude

/**
 * 오류 시 [ApiResponse.result] 에 담기는 본문.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
class ApiErrorBody private constructor(
    val message: String?,
    val messageCode: String?,
    val errors: Map<String, String>?,
) {
    companion object {
        fun of(messageCode: String?, message: String?, errors: Map<String, String>?): ApiErrorBody =
            ApiErrorBody(message, messageCode, errors)
    }
}
